Assets/Scripts/FeedingSystem/DogFoodBowl.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+
+public class DogFoodBowl : MonoBehaviourExtended
+{
+
+    #region Instance Accessors
+
+    public float FeedTimerLeft
+    {
+        get
+        {
+            return feedingTimer.TimeRemaining;
+        }
+    }
+
+    public bool IsCurrentlyFeeding
+    {
+        get
+        {
+            return feedingTimer.TimeRemaining != 0;
+        }
+    }
+
+    #endregion
+
+    static PPTimer feedingTimer = null;
+
+    protected override void fetchReferences() {
+        base.fetchReferences();
+
+        if (feedingTimer == null)
+        {
+            feedingTimer = new PPTimer(20, 1f);
+            feedingTimer.SetTimeRemaining(0, false);
+        }
+        feedingTimer.SubscribeToTimeUp(handleFeedingTimeUp);
+        GetComponent<Button>().interactable = !IsCurrentlyFeeding;
+    }
+
+    protected override void cleanupReferences()
+    {
+        base.cleanupReferences();
+        feedingTimer.UnsubscribeFromTimeUp(handleFeedingTimeUp);
+    }
+
+    int calculateDogFoodNeeded()
+    {
+        return PPDataController.GetInstance.DogCount - PPDataController.GetInstance.ScoutingDogs.Count;
```

Comment:
Feel like this should be a single function or property in PPDataController: 

e.g. `int DogsHomeCount` 

Also, you should save a class var for `PPDataController` if you're referencing it in multiple places, instead of reusing the singleton accessor, See structure in `PPGameController` among other classes